src/cljs/main/broadfcui/common/entity_table.cljs

```diff
@@ -26,8 +26,14 @@
     item))
 
 (react/defc EntityTable
-  {:refresh
-   (fn [{:keys [props state refs]} & [entity-type]]
+  {:update-data
+   (fn [{:keys [refs after-update]} reinitialize?]
+     (if reinitialize?
+       (do (react/call :refresh-rows (@refs "table"))
+           (after-update #(react/call :reinitialize (@refs "table"))))
```

Comment:
This `after-update` is incorrect, because it's waiting on `EntityTable`'s state, when we need to wait on `Table`'s state.  I tried to fix this by moving the call to `:reinitialize` into `:refresh-rows` (making it Table's responsibility), but there seems to still be a race condition going on.